pkg/ext/resources/tokens/store.go

```diff
@@ -0,0 +1,325 @@
+	authed, err := t.userHasFullPermissions(userInfo)
+	if err != nil {
+		return nil, fmt.Errorf("unable to check if user has * on tokens: %w", err)
+	}
+
+	ch := make(chan types.WatchEvent[*RancherToken])
+
+	go func() {
+		watcher, err := t.secretClient.Watch(TokenNamespace, metav1.ListOptions{
+			ResourceVersion: opts.ResourceVersion,
+		})
+		if err != nil {
+			return
+		}
+		defer watcher.Stop()
+
+		for event := range watcher.ResultChan() {
+			secret, ok := event.Object.(*corev1.Secret)
+			if !ok {
+				continue
+			}
+
+			token := tokenFromSecret(secret)
+			if !authed && token.Spec.UserID != userInfo.GetName() {
+				continue
+			}
+
+			watchEvent := types.WatchEvent[*RancherToken]{
+				Event:  event.Type,
+				Object: token,
+			}
+			ch <- watchEvent
+		}
+	}()
+
+	return ch, nil
+}
+
+func (t *TokenStore) Delete(ctx context.Context, userInfo user.Info, name string, opts *metav1.DeleteOptions) error {
+	fmt.Println(userInfo, name)
+	secret, err := t.secretCache.Get(TokenNamespace, name)
+	if err != nil {
+		return fmt.Errorf("unable to confirm secret existence %s: %w", name, err)
+	}
+	token := tokenFromSecret(secret)
+	// user is trying to get a token for another user
+	if token.Spec.UserID != userInfo.GetName() {
+		authed, err := t.userHasFullPermissions(userInfo)
+		if err != nil {
+			return fmt.Errorf("unable to check if user has * on tokens: %w", err)
+		}
+		if !authed {
+			return fmt.Errorf("can't create token for other user %s since user %s doesn't have * on ranchertokens", userInfo.GetName(), token.Spec.UserID)
+		}
+	}
+	err = t.secretClient.Delete(TokenNamespace, name, &metav1.DeleteOptions{})
+	if err != nil {
+		return fmt.Errorf("unable to delete secret %s: %w", name, err)
+	}
+	return nil
+}
+
+var _ types.TableConvertor[*RancherTokenList] = (*TokenStore)(nil)
+
+func (t *TokenStore) ConvertToTable(list *RancherTokenList, opts *metav1.TableOptions) *metav1.Table {
+	table := &metav1.Table{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Table",
+			APIVersion: "meta.k8s.io/v1",
+		},
+		ListMeta: list.ListMeta,
+		ColumnDefinitions: []metav1.TableColumnDefinition{
+			{
+				Name: "Name",
+				Type: "string",
+			},
+			{
+				Name: "Enabled",
+				Type: "boolean",
+			},
+			{
+				Name: "Cluster",
+				Type: "string",
+			},
+			{
+				Name:   "Age",
+				Type:   "string",
+				Format: "date",
+			},
+		},
+	}
+
+	for _, item := range list.Items {
+		row := metav1.TableRow{
+			Cells: []any{
+				item.Name,
+				item.Spec.Enabled,
+				item.Spec.ClusterName,
+				item.CreationTimestamp,
+			},
+		}
+		table.Rows = append(table.Rows, row)
+	}
+
+	return table
+}
+
+func (t *TokenStore) userHasFullPermissions(user user.Info) (bool, error) {
+	sar := &authv1.SubjectAccessReview{
+		Spec: authv1.SubjectAccessReviewSpec{
+			User:   user.GetName(),
+			Groups: user.GetGroups(),
+			UID:    user.GetUID(),
+			ResourceAttributes: &authv1.ResourceAttributes{
+				Verb:     "*",
```

Comment:
How would a regular user differ from the admin with full permissions ?
IOW which of the verbs will the regular user not have ?

Or is there a difference between `verb: "*"`, and `verb: [create,update,get,patch,watch,list,delete]`, i.e. having a `*` versus explicitly listing all the verbs?